From mano@roarinelk.homelinux.net Wed Feb 21 05:47:27 2007
From: Manuel Lauss <mano@roarinelk.homelinux.net>
Date: Wed, 21 Feb 2007 14:47:20 +0100
Subject: Driver core: more fallout from class_device changes for pcmcia
To: <linux-pcmcia@lists.infradead.org>
Cc: gregkh@suse.de
Message-ID: <20070221134720.GA10407@roarinelk.homelinux.net>
Content-Disposition: inline


More fallout from the PCMCIA class_device changes.

The first hunk is run-tested on SH-4, the others are converted
in the spirit of the original conversion.


Signed-off-by: Manuel Lauss  <mano@roarinelk.homelinux.net>
Signed-off-by: Greg Kroah-Hartman <gregkh@suse.de>


---
 drivers/pcmcia/hd64465_ss.c   |    2 +-
 drivers/pcmcia/m32r_cfc.c     |    2 +-
 drivers/pcmcia/m8xx_pcmcia.c  |    2 +-
 drivers/pcmcia/omap_cf.c      |    2 +-
 drivers/pcmcia/rsrc_mgr.c     |    2 +-
 drivers/pcmcia/vrc4171_card.c |    2 +-
 6 files changed, 6 insertions(+), 6 deletions(-)

--- gregkh-2.6.orig/drivers/pcmcia/hd64465_ss.c
+++ gregkh-2.6/drivers/pcmcia/hd64465_ss.c
@@ -907,7 +907,7 @@ static int __init init_hs(void)
 
 	for (i=0; i<HS_MAX_SOCKETS; i++) {
 		unsigned int ret;
-		hs_sockets[i].socket.dev.dev = &hd64465_device.dev;		
+		hs_sockets[i].socket.dev.parent = &hd64465_device.dev;
 		hs_sockets[i].number = i;
 		ret = pcmcia_register_socket(&hs_sockets[i].socket);
 		if (ret && i)
--- gregkh-2.6.orig/drivers/pcmcia/m32r_cfc.c
+++ gregkh-2.6/drivers/pcmcia/m32r_cfc.c
@@ -760,7 +760,7 @@ static int __init init_m32r_pcc(void)
 	/* Set up interrupt handler(s) */
 
 	for (i = 0 ; i < pcc_sockets ; i++) {
-		socket[i].socket.dev.dev = &pcc_device.dev;
+		socket[i].socket.dev.parent = &pcc_device.dev;
 		socket[i].socket.ops = &pcc_operations;
 		socket[i].socket.resource_ops = &pccard_nonstatic_ops;
 		socket[i].socket.owner = THIS_MODULE;
--- gregkh-2.6.orig/drivers/pcmcia/m8xx_pcmcia.c
+++ gregkh-2.6/drivers/pcmcia/m8xx_pcmcia.c
@@ -1321,7 +1321,7 @@ static int __init m8xx_init(void)
 		socket[i].socket.ops = &m8xx_services;
 		socket[i].socket.resource_ops = &pccard_iodyn_ops;
 		socket[i].socket.cb_dev = NULL;
-		socket[i].socket.dev.dev = &m8xx_device.dev;
+		socket[i].socket.dev.parent = &m8xx_device.dev;
 	}
 
 	for (i = 0; i < PCMCIA_SOCKETS_NO; i++)
--- gregkh-2.6.orig/drivers/pcmcia/omap_cf.c
+++ gregkh-2.6/drivers/pcmcia/omap_cf.c
@@ -291,7 +291,7 @@ static int __devinit omap_cf_probe(struc
 		omap_cf_present() ? "present" : "(not present)");
 
 	cf->socket.owner = THIS_MODULE;
-	cf->socket.dev.dev = dev;
+	cf->socket.dev.parent = dev;
 	cf->socket.ops = &omap_cf_ops;
 	cf->socket.resource_ops = &pccard_static_ops;
 	cf->socket.features = SS_CAP_PCCARD | SS_CAP_STATIC_MAP
--- gregkh-2.6.orig/drivers/pcmcia/rsrc_mgr.c
+++ gregkh-2.6/drivers/pcmcia/rsrc_mgr.c
@@ -232,7 +232,7 @@ static struct resource *iodyn_find_io_re
 		unsigned long align, struct pcmcia_socket *s)
 {
 	struct resource *res = make_resource(0, num, IORESOURCE_IO,
-					     s->dev.class_id);
+					     s->dev.bus_id);
 	struct pcmcia_align_data data;
 	unsigned long min = base;
 	int ret;
--- gregkh-2.6.orig/drivers/pcmcia/vrc4171_card.c
+++ gregkh-2.6/drivers/pcmcia/vrc4171_card.c
@@ -596,7 +596,7 @@ static int __devinit vrc4171_add_sockets
 		}
 
 		sprintf(socket->name, "NEC VRC4171 Card Slot %1c", 'A' + slot);
-		socket->pcmcia_socket.dev.dev = &vrc4171_card_device.dev;
+		socket->pcmcia_socket.dev.parent = &vrc4171_card_device.dev;
 		socket->pcmcia_socket.ops = &vrc4171_pccard_operations;
 		socket->pcmcia_socket.owner = THIS_MODULE;
 
